### PR TITLE
Move git diff codes from models to services/gitdiff

### PR DESCRIPTION
**Pull request by lunny**
As title.

### CODE REVIEW

**Review by typeless — COMMENTED** [review]
What is the main difference in purposes between `modules` and `services` directories?

**Comment by lunny** [comment]
`services` will depend on `modes`/`modules` and be dependent by `routers`/`commands`.

File: services/comments/comments.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	"fmt"
+	"strings"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/services/gitdiff"
+)
+
+// CreateCodeComment creates a plain code comment at the specified line / path
```
[the remark this answers is not in the document]
Comment (lunny) [reply services/comments/comments.go:18]: I would like to leave it for another PR.

[the remark this answers is not in the document]
Comment (typeless) [reply services/comments/comments.go:18]: No problem.

File: services/comments/comments.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		return nil, fmt.Errorf("GetHeadRepo: %v", err)
+	}
+	gitRepo, err := git.OpenRepository(pr.BaseRepo.RepoPath())
+	if err != nil {
+		return nil, fmt.Errorf("OpenRepository: %v", err)
+	}
+
+	// FIXME validate treePath
+	// Get latest commit referencing the commented line
+	// No need for get commit for base branch changes
```
[the remark this answers is not in the document]
Comment (lunny) [reply services/comments/comments.go:35]: Just move them and haven't see it's correct.

File: services/comments/comments.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	if line > 0 {
+		commit, err := gitRepo.LineBlame(pr.GetGitRefName(), gitRepo.Path, treePath, uint(line))
+		if err == nil {
+			commitID = commit.ID.String()
+		} else if !strings.Contains(err.Error(), "exit status 128 - fatal: no such path") {
+			return nil, fmt.Errorf("LineBlame[%s, %s, %s, %d]: %v", pr.GetGitRefName(), gitRepo.Path, treePath, line, err)
+		}
+	}
+
+	// Only fetch diff if comment is review comment
```
[the remark this answers is not in the document]
Comment (lunny) [reply services/comments/comments.go:45]: Just move them and haven't see it's correct.

**Review by typeless — APPROVED** [review]

File: services/comments/comments.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (typeless) [reply services/comments/comments.go:18]: @lunny By the way, after realizing it's already there and this is just a refactor. I think it is not worth the effort to change the name at this point.

**Review by techknowlogick — APPROVED** [review]

**Review by guillep2k — APPROVED** [review]

**Comment by techknowlogick** [comment]
ping LG-TM bot

**Pull request merged**